File: app/src/main/java/tech/pauly/findapet/shared/events/FragmentEvent.java
package tech.pauly.findapet.shared.events;

import android.support.annotation.IdRes;

public class FragmentEvent extends BaseViewEvent {

    private Class fragmentClass;
    @IdRes
    private int containerId;

    public static FragmentEvent build(Object emitter) {
        return new FragmentEvent(emitter.getClass());
    }

    private FragmentEvent(Class emitter) {
        this.emitter = emitter;
    }

    public FragmentEvent fragment(Class fragmentClass) {
        this.fragmentClass = fragmentClass;
        return this;
    }

    public FragmentEvent container(@IdRes int id) {
        this.containerId = id;
        return this;
    }

    public Class getFragmentClass() {
        return fragmentClass;
    }

    public int getContainerId() {
        return containerId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        FragmentEvent that = (FragmentEvent) o;

        if (containerId != that.containerId) {
            return false;
        }
        return fragmentClass != null ? fragmentClass.equals(that.fragmentClass) : that.fragmentClass == null;
    }

    @Override
    public int hashCode() {
        int result = fragmentClass != null ? fragmentClass.hashCode() : 0;
        result = 31 * result + containerId;
        return result;
    }
}

File: app-old/src/main/java/tech/pauly/old/shared/events/FragmentEvent.kt
package tech.pauly.old.shared.events

import androidx.annotation.IdRes
import kotlin.reflect.KClass

open class FragmentEvent(private val emitter: Any,
                         open val fragment: KClass<*>,
                         @IdRes open val container: Int) : BaseViewEvent(emitter::class) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as FragmentEvent

        if (emitter != other.emitter) return false
        if (fragment != other.fragment) return false
        if (container != other.container) return false

        return true
    }

    override fun hashCode(): Int {
        var result = emitter.hashCode()
        result = 31 * result + fragment.hashCode()
        result = 31 * result + container
        return result
    }
}